### PR TITLE
Use New Permissions Framework on Annotations Handler 

### PR DESCRIPTION
**Pull request by dannygoldstein**
Updates the annotations handler to use the permissions checking infrastructure introduced in #1542

### CODE REVIEW

**Review by dmitryduev — APPROVED** [review]
So much cleaner!

File: skyportal/handlers/api/annotation.py [hunk 1]
```diff
@@ -110,51 +115,6 @@ def post(self):
             return self.error("Input `origin` must begin with alphanumeric/underscore")
 
         annotation_data = data.get("data")
-
-        # Ensure user/token has access to parent source
```
Comment (guynir42) [inline skyportal/handlers/api/annotation.py:114]: So what happens if the user doesn't have access to the source? it fails upon trying to post to the database?

**Review by guynir42 — APPROVED** [review]
I agree, this is a lot more concise... I want to go and see where this magic happens, is it somewhere inside baselayer?

File: skyportal/handlers/api/annotation.py (hunk 1 above)
Comment (dannygoldstein) [reply skyportal/handlers/api/annotation.py:114]: Yes

**Comment by dannygoldstein** [comment]
@guynir42 it happens here - 

https://github.com/skyportal/skyportal/blob/master/skyportal/models.py#L2059
https://github.com/cesium-ml/baselayer/blob/master/app/models.py#L288
https://github.com/cesium-ml/baselayer/blob/master/app/handlers/base.py#L84

**Comment by dannygoldstein** [comment]
Failure unrelated - was flaky on #1542

**Pull request merged**